src/WebJobs.Script.WebHost/Controllers/KeysController.cs

```diff
@@ -285,9 +287,13 @@ internal bool IsBuiltInSystemKeyName(string keyName)
 
         private bool IsFunction(string functionName)
         {
-            string json = null;
+            if (_functionMetadataManager.TryGetFunctionMetadata(functionName, out _))
+            {
+                return true;
+            }
```

Comment:
I'm wondering if you should flip this to have less impact. Try to read the FunctionConfig first, and then, if false, try this as a fallback? 

I don't know what could break from this, but it's definitely a behavior change like this if the function.json is present.

Reply:
Agreed. The previous behavior will actually attempt to parse the file. Placing this here does have a behavior impact.

Reply:
Admin APIs like this have to work even if the host isn't running (i.e. keys/functions APIs aren't marked RequiresRunningHost). Doesn't the metadata manager require host startup to initialize?

Reply:
Yeah I will flip this. @mathewc, metadata manager doesn't rely on Host running. When Host starts, it reinitializes the list of functions it has (including codeless ones). But it shouldn't block on anything like that.

Reply:
Ok